### PR TITLE
Partner

### PR DESCRIPTION
**Pull request by KMayuri25**

### CODE REVIEW

File: src/app/partner/update-partner/update-partner.component.ts [hunk 1]
```diff
@@ -80,29 +147,34 @@ export class UpdatePartnerComponent implements OnInit {
...
     } else {
+        this.partnerObject.name = this.control.partner_name.value;
+        this.partnerObject.buId = this.control.business_unit.value;
+        this.partnerObject.username = this.control.username.value;
+        this.partnerObject.password = this.control.password.value;
+        this.partnerObject.config = (JSON.stringify(this.config_val_fz = { 'appToken': this.registerForm.get('app_token').value, 'wardenAPIURI': this.registerForm.get('warden_authURI').value,  'bossAPIURI': this.registerForm.get('bossAPI').value}));
+        this.partnerObject.email = this.control.email.value;
+        this.partnerObject.status = 'active';
+
+      console.log(this.partnerObject);
```
Comment (pooja-koul-incontact) [inline src/app/partner/update-partner/update-partner.component.ts]: remove logger

File: src/app/partner/update-partner/update-partner.component.ts [hunk 2]
```diff
@@ -80,29 +147,34 @@ export class UpdatePartnerComponent implements OnInit {
...
-        this.showErrorMsg = true;
-      });
+    this.partnerService.update(this.id, this.partnerObject).subscribe((response: Response) => {
+      this.router.navigate(['partner']);
+    }, error => {
+      localStorage.setItem('ErrorMsg', error );
+      this.showErrorMsg = true;
+    });
+
+    console.log(this.partnerObject);
```
Comment (pooja-koul-incontact) [inline src/app/partner/update-partner/update-partner.component.ts]: remove logger

File: src/app/partner/update-partner/update-partner.component.ts [hunk 3]
```diff
@@ -80,29 +147,34 @@ export class UpdatePartnerComponent implements OnInit {
     this.partnerObject.email = this.control.email.value;
     this.partnerObject.status = 'active';
 
+      console.log(this.partnerObject);
```
Comment (pooja-koul-incontact) [inline src/app/partner/update-partner/update-partner.component.ts]: remove here also

File: src/app/partner/create-partner/create-partner.component.ts [hunk 4]
```diff
@@ -47,31 +120,64 @@ export class CreatePartnerComponent implements OnInit {
...
+        'username': this.control.username.value,
+        'password': this.control.password.value,
+         'config' : (JSON.stringify(this.config_val_rc = { 'baseURI': this.control.base_uri.value, 'secret': this.control.client_secret.value, 'authURI': this.control.auth_uri.value})),
+        'email' : this.control.email.value,
+        'status' : 'active',
+        'format' : '1'
+      };
+
+
+      console.log(this.partnerObject);
```
Comment (pooja-koul-incontact) [inline src/app/partner/create-partner/create-partner.component.ts]: remove

File: src/app/partner/create-partner/create-partner.component.ts [hunk 5]
```diff
@@ -47,31 +120,64 @@ export class CreatePartnerComponent implements OnInit {
...
+        'name': this.control.partner_name.value,
+        'email' : this.control.email.value,
+        'username': this.control.username.value,
+        'password': this.control.password.value,
+        'config' : (JSON.stringify(this.config_val_fz = { 'appToken': this.registerForm.get('app_token').value, 'wardenAPIURI': this.registerForm.get('warden_authURI').value,  'bossAPIURI': this.registerForm.get('bossAPI').value})),
+        'status' : 'active',
+        'format': '2'
+      };
+
+      console.log(this.partnerObject);
```
Comment (Susmith-roy-incontact) [inline src/app/partner/create-partner/create-partner.component.ts]: remove logger

File: src/app/partner/create-partner/create-partner.component.ts [hunk 6]
```diff
@@ -47,31 +120,64 @@ export class CreatePartnerComponent implements OnInit {
...
+      this.partnerService.create(this.partnerObject).subscribe((res: Response) => {
+        this.router.navigate(['partner']);
+      }, error => {
+        localStorage.setItem('ErrorMsg', error );
```
Comment (Susmith-roy-incontact) [inline src/app/partner/create-partner/create-partner.component.ts]: Delete after use

File: src/app/partner/create-partner/create-partner.component.ts [hunk 7]
```diff
@@ -47,31 +120,64 @@ export class CreatePartnerComponent implements OnInit {
...
+        'status' : 'active',
+        'format' : '1'
+      };
+
+
+      console.log(this.partnerObject);
+      this.partnerService.create(this.partnerObject).subscribe( res => {
+        this.router.navigate(['partner']);
+      }, (error: HttpErrorResponse) => {
+        localStorage.setItem('ErrorMsg', error.error.errors[0].message );
```
Comment (Susmith-roy-incontact) [inline src/app/partner/create-partner/create-partner.component.ts]: remove after use

File: src/app/partner/update-partner/update-partner.component.ts [hunk 8]
```diff
@@ -15,40 +14,107 @@ import {ConfigData} from './model/configData';
...
+      bossAPI : [null],
+      warden_authURI : [null],
+      auth_uri: ['', [Validators.required , Validators.pattern(/^(?:([A-Za-z]+):)?(\/{0,3})([0-9.\-A-Za-z]+)(?::(\d+))?(?:\/([^?#]*))?(?:\?([^#]*))?(?:#(.*))?$/)]],
       base_uri: ['', [Validators.required , Validators.pattern(/^(?:([A-Za-z]+):)?(\/{0,3})([0-9.\-A-Za-z]+)(?::(\d+))?(?:\/([^?#]*))?(?:\?([^#]*))?(?:#(.*))?$/)]],
-      client_secret: ['', [Validators.required, Validators.pattern(/^[a-zA-Z0-9=]+$/)]]
+      client_secret: ['', [Validators.required, Validators.pattern(/^[a-zA-Z0-9=]+$/)]],
+
     });
+
+    console.log(this.partnerObject.format);
```
Comment (Susmith-roy-incontact) [inline src/app/partner/update-partner/update-partner.component.ts]: remove

File: src/app/partner/update-partner/update-partner.component.ts [hunk 9]
```diff
@@ -15,40 +14,107 @@ import {ConfigData} from './model/configData';
...
+    const base_uri = this.registerForm.get('base_uri');
+    const client_secret = this.registerForm.get('client_secret');
+    const app_token = this.registerForm.get('app_token');
+    const bossAPI = this.registerForm.get('bossAPI');
+    const warden_authURI = this.registerForm.get('warden_authURI');
+
+    console.log(this.registerForm.get('userCategory').value);
+    this.registerForm.get('userCategory').valueChanges
+      .subscribe(userCategory => {
+        console.log("Hello");
```
Comment (Susmith-roy-incontact) [inline src/app/partner/update-partner/update-partner.component.ts]: remove

File: src/app/partner/update-partner/update-partner.component.ts [hunk 10]
```diff
@@ -80,29 +147,34 @@ export class UpdatePartnerComponent implements OnInit {
...
-          this.router.navigate(['partner']);
-        });
-      }, error => {
-        localStorage.setItem('ErrorMsg', error );
-        this.showErrorMsg = true;
-      });
+    this.partnerService.update(this.id, this.partnerObject).subscribe((response: Response) => {
+      this.router.navigate(['partner']);
+    }, error => {
+      localStorage.setItem('ErrorMsg', error );
```
Comment (Susmith-roy-incontact) [inline src/app/partner/update-partner/update-partner.component.ts]: remove after use

File: src/app/partner/update-partner/update-partner.component.ts [hunk 11]
```diff
@@ -80,29 +147,34 @@ export class UpdatePartnerComponent implements OnInit {
...
+        this.partnerObject.email = this.control.email.value;
+        this.partnerObject.status = 'active';
+
+      console.log(this.partnerObject);
 
       this.partnerService.update(this.id, this.partnerObject).subscribe((response: Response) => {
         this.router.navigate(['partner']);
-      },error => {
+      }, error => {
         localStorage.setItem('ErrorMsg', error );
```
Comment (Susmith-roy-incontact) [inline src/app/partner/update-partner/update-partner.component.ts]: remove after use

File: src/app/partner/update-partner/update-partner.component.ts [hunk 12]
```diff
@@ -15,40 +14,107 @@ import {ConfigData} from './model/configData';
...
+  }
+
+  ngOnInit() {
+
+    this.partnerObject = JSON.parse(localStorage.getItem('ClickedRowData'));
+    if (this.partnerObject.format === '1') {
+      this.config_val = JSON.parse(this.partnerObject.config || '{}');
+    } else {
+      this.config_val_fz = JSON.parse(this.partnerObject.config || '{}');
+      console.log(this.config_val_fz);
```
Comment (Susmith-roy-incontact) [inline src/app/partner/update-partner/update-partner.component.ts]: Remove

File: src/app/partner/update-partner/update-partner.component.ts [hunk 13]
```diff
@@ -15,40 +14,107 @@ import {ConfigData} from './model/configData';
...
+        bossAPI.updateValueAndValidity();
+        warden_authURI.updateValueAndValidity();
+
+      });
+  }
+
+  ngOnInit() {
+
+    this.partnerObject = JSON.parse(localStorage.getItem('ClickedRowData'));
+    if (this.partnerObject.format === '1') {
```
Comment (Susmith-roy-incontact) [inline src/app/partner/update-partner/update-partner.component.ts]: If and else block is doing same task, do we need this if else block

File: src/app/partner/update-partner/update-partner.component.ts [hunk 14]
```diff
@@ -15,40 +14,107 @@ import {ConfigData} from './model/configData';
...
+      this.config_val = JSON.parse(this.partnerObject.config || '{}');
+    } else {
+      this.config_val_fz = JSON.parse(this.partnerObject.config || '{}');
+      console.log(this.config_val_fz);
+    }
+
+    this.id = JSON.parse(localStorage.getItem('ClickedRowData')).uuid ;
```
Comment (Susmith-roy-incontact) [inline src/app/partner/update-partner/update-partner.component.ts]: remove after use

**Pull request merged**